server/routes/deck.js

```diff
@@ -0,0 +1,73 @@
+const router = require('express').Router();
+const mongoose = require('mongoose');
+require('../models/DeckSchema');
+const Deck = mongoose.model('Deck');
+
+router.post('/createDeck', function(req, res) {
+    const deck = new Deck();
+    deck.title = req.body.title;
+    
+    let shortId;
+    do {
+        shortId = Math.random().toString(36).substr(2, 6).toUpperCase()
+    } while(Deck.findOne({'shortId': shortId}).length);
```

Comment:
Generalnie nie robiłbym tego w pętli, ponieważ findOne da ci tylko 1 wynik. Najlepiej przerobić to na obsługe promise, bo obecnie nie obsługujesz żadnego błędu,  kiedy funkcja findOne nic nie znajdzie. Czyli przerobiłbym to na findone().then().catch()